### ISSUE TITLE
Form input visuals

### ISSUE THREAD

**Issue by joseegm** [SAP/techne#737]
### Goals
1. Create visual styles for all form input fields listed on #733 
2. Create visual styles for all status from #736 
3. Create visual styles for all help elements #738 

### Current pain points to address:
1. Label font is very small in relation to the input line.
2. Colors of the labels are not contrasting enough.
3. Input lines looks too big in relation to the labels.
4. Faux 3d emboss effect on the input lines - not needed.

### Prior art as v1.5.x
![screen shot 2017-06-19 at 11 34 32](https://user-images.githubusercontent.com/22662903/27278972-4e2018a8-54e3-11e7-82a5-5b4f293ce54a.png)

**Comment by joseegm** [comment]
### Starting Draft
![input-lines 1](https://user-images.githubusercontent.com/22662903/27283124-22c21b10-54f3-11e7-946e-e6c2dc0a1fd2.png)

**Comment by joseegm** [comment]
@LeoT7508 
You haven't post the visuals you show yesterday. So I cannot take a detailed look at it.

We can agree to post any progress as it is available. That will speed up things and make the improvements based on the feedback visible.

From what I saw yesterday:

### Labels
1. They are still small compared with the font inside the input. Ideally they will have the same font-size with the font inside the input having slightly more visual weight than the label.
2. Color is not contrasting enough

### Inputs
1. The current border looks very rigid we can "soft" it a bit if we try rounding the corners. Like
  ![form input elements](https://user-images.githubusercontent.com/22662903/27285538-55f17ae4-54fd-11e7-9a75-3cf95dcc44df.png)
2. The extra elements: **actions**, **sping box** have their own border. Let's try having a contrasting background and using spacing to separate it from the input.

**Comment by LeoT7508** [comment]
@joseegm 

I have not posted the visuals as those are not ready yet. I have to make them pixel perfect, and they aren't at that point yet. 

To your other questions:

Labels:

1. In general the labels and the text in the fields should not carry the same weight or size, there needs to be a contrast in size and weight to show hierarchy and priority. That being said labels aren't the primary action in forms, its the information the user puts in the field. With all that taken into consideration the labels will be 13px bold. And the text inside the box will be our normal body copy, 16px regular. 

The colors I'm using pass accessibility standards, so we should be good there. 

Inputs

1. Im going to stay away from rounding corners on fields to keep things consistent with other elements we will need to add later, like buttons, tool and action bars. Buttons that live in the tool bar are squared. Id rather not have variations in style per component. Keeping them square will allow us more flexibility. 

2. The extra elements like actions, and spring box will not have their own border, this hints at a button or action that needs to be clicked, when in reality they aren't. Visually I would like to keep the info in the fields without separation. We will create the space you are asking for using color with icons or a solid background for buttons when needed. This all will reduce confusion and makes it very clear whats happening. 

Im close to being done with these visual tasks. Its taking more time because I have to start accounting for buttons and icons, so I'm designing these 3 at the same time. I have to make sure everything is pixel perfect or it will come back to bite us later on. 

Thanks for the patience.

**Comment by LeoT7508** [comment]
@joseegm 

![techne_2 0_inputfields](https://user-images.githubusercontent.com/28761982/27450154-527670ae-5750-11e7-9f62-eaf6fa5789a1.jpg)

**Comment by joseegm** [comment]
Regarding points on https://github.com/SAP/techne/issues/737#issuecomment-310412094

**Labels**
1. I understand the points and agree with them. But the final result look disproportionated. Visually the input is way "heavier" than the label.
  ![27450154-527670ae-5750-11e7-9f62-eaf6fa5789a1 1](https://user-images.githubusercontent.com/22662903/27453963-7e259ade-5798-11e7-8472-f810984d6d62.jpg)
  It is probably the padding on the input.

    Looking other libraries:
    [Fabric](https://dev.office.com/fabric#Variants): Have bigger label font size than input font size.
    [Blueprint](http://blueprintjs.com/docs/#core/components/forms/label.simple-labels): Have the same size.

     **We need to test this to be 100% sure**, testing will be done at #752 -I'll ping you over there to see the details
   Just to be completely impartial and see what the wild user understands better.

    Since that test involves only one variable (font-size) that we can change later. We can move on with this font size.

**Inputs**
1. Okay.
2. Got it.

### Status
*Cropping the image for context*
![27450154-527670ae-5750-11e7-9f62-eaf6fa5789a1](https://user-images.githubusercontent.com/22662903/27453818-f4f44904-5797-11e7-82e8-c9529e9509f6.jpg)

1. Still missing the **Valid** Status
2. Still missing **Mandatory**. Define on https://github.com/SAP/techne/wiki/Form-Inputs#form-inputs-status-763 mandatory's description. The mandatory field should have a bigger contrast than the rest.
  So the user recognizes immediately that field is important. Not only for the `*` on the label.

**Comment by jeannewalters** [comment]
@joseegm @LeoT7508 Can you please add Hana to this ticket as well? For Valid Status, can you add more context around how this would be used? I was talking to Leo about this today and I would ask that you clarify in this ticket, what you are considering to be the use case for a "valid" field. I was trying to help him from the UX side but we aren't expecting teams to do inline validation or validation of focus for every field - sometimes, there would be the case where data in fields aren't validated until the primary button is clicked. In order to better guide him from a UX perspective, I need to understand the use case from you. I would also suggest you submit research tickets through the survey monkey link she sent out a couple weeks ago so that she can give you the best practice for anything that may be outside of the prototype link I sent you with Irla's wires.

**Comment by LeoT7508** [comment]
@joseegm 

I reached out to Jeanne and Irla about UX feedback and research that was done with fields in SmartEdit. 

1. I will wait to see about Validation per Jeanne's comments above. 

2. I had to make a cross between SmartEdit & Techne to update the field labels. In Techne the field label weight is regular, in SmartEdit they are bold. So for normal field labels I used Semibold and for mandatory fields I used Bold. 

3. I also updated the Error state per the user feedback from SmartEdit. 

![techne_2 0_inputfields](https://user-images.githubusercontent.com/28761982/27490044-74b2cc28-5802-11e7-9f0e-31d45a4625ec.jpg)

**Comment by joseegm** [comment]
@jeannewalters 
TestTest was already requested to Hanna is already, see #752. If you need something else tested please open an issue with the test details like #757. And post the results as soon as you have it. Tests and test results have to be visible.

**Comment by joseegm** [comment]
@LeoT7508 
1. No need to wait. The Valid status only highlight the Form Input element. Most used method is a green border. It is the opposite to the **Invalid** Status. 
2. Semibold is okay of design, but it will not be much noticeable in the browser. We can go for **Regular** for normal and **Bold** for mandatory.
3. That design makes more clear that the errors belong to the input field.

Only thing missing is the **Valid** status.

**Comment by LeoT7508** [comment]
@joseegm 

1. I have a green color we can use for "Valid" states once Jeanne's concerns are addressed. 

2. I updated the labels.

**Comment by joseegm** [comment]
@LeoT7508 
1. Jeanne's concerns where answered on https://github.com/SAP/techne/issues/737#issuecomment-310976692. If she want to do some extra tests, it will be posted here.
  I think we have enough elements here to move on with the **Valid** status. 
  
2. Very good. Can you please post the updated visuals here? To have more context. Thank you.

**Comment by jeannewalters** [comment]
@joseegm - I am asking you to be clear on what you are asking for this field type. Is this only the case for forms where a person is logging and and validation is done immediately for this field?  
You are not explaining the use case for these before asking Leo to design it visually - only adding labels. That's not ok in my UX book.
Jose said:
Focus and Validation: Don't understand what you are trying to say. Maybe rephrase that in a more clear language.

**Comment by joseegm** [comment]
@jeannewalters 
Thanks for rephrasing your comment. It is more clear now.

The **Valid** state can be used also for validating emails and other fields in realtime. As I already mentioned in https://github.com/SAP/techne/issues/737#issuecomment-310976692

**Comment by LeoT7508** [comment]
@joseegm 

![techne_2 0_inputfields](https://user-images.githubusercontent.com/28761982/27642792-731aae9e-5be5-11e7-85f9-238d99f1b30f.jpg)
![techne_2 0_inputs types with common elements](https://user-images.githubusercontent.com/28761982/27642793-732507b8-5be5-11e7-815f-5c094cb8219a.jpg)

**Comment by joseegm** [comment]
@LeoT7508 
Can you update the visuals again. To finish this task.

Changes missing:
1. Valid state: green frame
2. Read only field with the line at the bottom (on second screenshot)
3. Action Button with same color as the action icons

Also please update the file on Box.

Thank you

**Comment by LeoT7508** [comment]
@joseegm 

![techne_2 0_inputfields](https://user-images.githubusercontent.com/28761982/27699293-63e0ab56-5cbf-11e7-98ab-c0aa8fb3549a.jpg)
![techne_2 0_inputs types with common elements](https://user-images.githubusercontent.com/28761982/27699294-63e26996-5cbf-11e7-91bb-a246f81c6c9b.jpg)

**Comment by joseegm** [comment]
@LeoT7508 Thanks
Still missing point 3. 
Or there is a reason that's not been changed

**Comment by LeoT7508** [comment]
@joseegm 

I updated the JPEG about with the colored button you asked for.
